File: tic-tac-toe-checker/tic-tac-toe-checker.py
import re
def isSolved(board):
    b = 'z'.join(''.join(str(j) for j in i)  for i in board)
    b1 = 'z'.join(''.join(str(j) for j in i)  for i in zip(*board))
    for i in {'1','2'}:
        rex = i+'{3}'
        if re.search(rex,b) or b[0]==b[5]==b[10]==i or b[2]==b[5]==b[8]==i or re.search(rex,b1):
            return int(i)
    return -1 if '0' in b  else 0 

File: tic-tac-toe-checker/tic-tac-toe-checker.js
function isSolved(board) {
 var board1 = board.map(a=>a.join("")).join("-")
 var test = board1.match(/([12])\d{2}-\1\d{2}-\1\d{2}|\d([12])\d-\d\1\d-\d\1\d|\d{2}([12])-\d{2}\1-\d{2}\1/)
 if(test){return parseInt(test[1])}
 var test2 = board1.match(/([12])\1{2}/)
 if(test2){return parseInt(test2[1])}
 var test3 = board1.match(/([12])\d{2}-\d\1\d-\d{2}\1|d{2}([12])-\d\2\d-\d\d\2/)
 if(test3){return parseInt(test3[1])}
 return board1.indexOf("0")==-1?0:-1
}